Add formatters for core log components (#775)
pinojs/pino: docs/api.md

* See the [redaction ⇗](/docs/redaction.md) documentation.
* See [fast-redact#caveat ⇗](http://github.com/davidmarkclements/fast-redact#caveat)

<a id=opt-formatters></a>
#### `formatters` (Object)

> An object containing functions for formatting the shape of the logs.

Reviewer: ```suggestion
An object containing functions for formatting the shape of the log lines.
```

> These functions should return an JSONifiable object and they

Reviewer: ```suggestion
These functions should return a JSONifiable object and
```

> should never throw. You can use this functions to shape the final JSON log in the way you need. For example you can use it for changing the level key name or enrich the default bindings.

Reviewer: ```suggestion
should never throw. These functions allow for full customization of
the resulting log lines. For example, they can be used to change
the level key name or to enrich the default metadata.
```

##### `level`

Changes the shape of the log level. The default shape is `{ level: number }`.

> The function takes two arguments, the label of the level (eg `'info'`) and the numeric value (eg: `30`).

Reviewer: ```suggestion
The function takes two arguments, the label of the level (e.g. `'info'`)
and the numeric value (e.g. `30`).
```

```js
const formatters = {
  level (label, number) {
    return { level: number }
  }
}
```

##### `bindings`

Changes the shape of the bindings. The default shape is `{ pid, hostname }`.

> The function takes a single argument which is the bindings object. It will be called every time you create a child logger.

Reviewer: ```suggestion
The function takes a single argument, the bindings object. It will
be called every time a child logger is created.
```

```js
const formatters = {
  bindings (bindings) {
    return { pid: bindings.pid, hostname: bindings.hostname }
  }
}
```

##### `log`

> Changes the shape of the log object. This function will be called every time one of the log methods (such as `.info`) is called with all the arguments passed to the log method but the message. By default it does not change the shape of the log object.

Reviewer: ```suggestion
Changes the shape of the log object. This function will be called every time
one of the log methods (such as `.info`) is called. All arguments passed to the
log method, except the message, will be pass to this function. By default it does
not change the shape of the log object.
```

* See [pino.stdSerializers](#pino-stdserializers)

> 

Reviewer 2: I think this is missing the `DEPRECATED` block
